presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,549 @@ Tuning Presto
+
+
+``query.max-memory-per-node``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``JVM max memory`` * ``0.1``
+ * **Description:** The purpose of that is same as of :ref:`query.max-memory<query-max-memory>` but the memory is not counted cluster-wise but node-wise instead.
+
+
+``query.max-queued-queries``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``5000``
+ * **Description:** **Deprecated** Describes how many queries may wait in worker queue. If the limit is reached master server will consider worker blocked and will not push more tasks to him. Setting this value high may allow to order a lot of queries at once with the cost of additional memory needed to keep informations about tasks to process. Lowering this value will decrease system capacity but will allow to utilize memore for real processing of date instead of queuing. It shouldn't be used in new configuration, the ``query.queue-config-file`` can be used instead.
+
+
+``query.max-run-time``
+^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (duration)
+ * **Default value:** ``100 d``
+ * **Description:** Used as default for session property ``query_max_run_time``. If the presto works in environment where there are mostly very long queries (over 100 days) than it may be a good idea to increase this value to avoid dropping clients that didn't set their session property correctly. On the other hand in the presto works in environment where they are only very short queries this value set to small value may be used to detect user errors in queries. It may also be decreased in poor presto cluster configuration with mostly short queries to increase garbage collection efficiency and by that lowering memory usage in cluster.
+
+
+``query.queue-config-file``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String``
+ * **Default value:**
+ * **Description:** This property may be defined to provide patch to queue config file. This is new way of providing such informations as ``query.max-concurrent-queries`` and ``query.max-queued-queries``. The file should contain JSON configuration described in :doc:`/admin/queue`.
```

Comment:
The path to the queue config file.  Queues are used to manage the number of concurrent queries across the system.  More information on queues and how to configure them can be found in :doc:`/admin/queue`.